fossunited/foss_hackathon/doctype/foss_hackathon_participant/foss_hackathon_participant.py

```diff
@@ -28,3 +28,18 @@ class FOSSHackathonParticipant(Document):
         wants_to_attend_locally: DF.Check
     # end: auto-generated types
     pass
+
+    def before_save(self):
+        self.handle_localhost_rejection()
+        if self.has_value_changed("localhost"):
+            self.update_request_status()
+
+    def update_request_status(self):
+        self.localhost_request_status = "Pending"
+
+    def handle_localhost_rejection(self):
+        if (
+            not self.has_value_changed("localhost")
+            and self.localhost_request_status == "Rejected"
+        ):
+            self.wants_to_attend_locally = 0
```

Comment:
nit - if we're setting a boolean flag, let's use a boolean and not something that is false-y

```suggestion
            self.wants_to_attend_locally = False
```